src/dbnode/storage/bootstrap.go

```diff
@@ -142,11 +145,18 @@ func (m *bootstrapManager) Bootstrap() (BootstrapResult, error) {
 		m.state = Bootstrapping
 	}
 	m.Unlock()
-
 	// NB(xichen): disable filesystem manager before we bootstrap to minimize
 	// the impact of file operations on bootstrapping performance
 	m.mediator.DisableFileOpsAndWait()
-	defer m.mediator.EnableFileOps()
+	defer func() {
+		if m.pOpts.BootstrapProfileEnabled() {
+			if err := profiler.WriteHeapProfile(m.pOpts.ProfilePath(),
```

Comment:
Shouldn't we be taking heap profiles at the same locations where we stop CPU profile?
Note that it is possible to see a diff of two heap profiles, so taking a separate one at the end of both phases might be useful.